src/app/shared/date.service.ts

```diff
@@ -73,4 +73,22 @@ export class DateService {
       return 'n/a';
     }
   }
+
+  getISO8601FormatFromDate(timestamp: Date): string {
+    if (timestamp) {
+      const tsAsDate = new Date(timestamp);
+      return tsAsDate.toISOString();
+    } else {
+      return 'n/a';
+    }
+  }
+
+  getISO8601FormatFromNumber(timestamp: number): string {
+    if (timestamp) {
```

Comment:
This should call `getISO8601FormatFromDate` with a date object so the (small) amount of code is is not duplicated.